build.gradle.kts

```diff
@@ -26,17 +28,90 @@ tasks.register<Delete>("clean") {
     delete(rootProject.layout.buildDirectory)
 }
 
+// Git Hooks Setup - Project-wide development tooling
+tasks.register("setupGitHooks") {
+    description = "Configure Git to use hooks from scripts directory"
+    group = "setup"
+
+    doLast {
+        // Check current hooks path configuration
+        val currentHooksPath = try {
+            val output = ByteArrayOutputStream()
+            exec {
+                commandLine("git", "config", "--get", "core.hooksPath")
+                standardOutput = output
+                isIgnoreExitValue = true
+            }
+            output.toString().trim()
+        } catch (e: Exception) {
+            ""
+        }
+
+        when {
+            currentHooksPath == "scripts" -> {
+                println("✅ Git hooks already configured correctly")
+            }
+            currentHooksPath.isEmpty() -> {
+                // Configure Git to use scripts directory for hooks
+                exec {
+                    commandLine("git", "config", "core.hooksPath", "scripts")
+                }
+                println("🔧 Configured Git to use scripts/ directory for hooks")
+            }
+            else -> {
+                println("⚠️ Git hooks path is currently set to: $currentHooksPath")
+                println("💡 To use project hooks, run: git config core.hooksPath scripts")
+            }
+        }
+
+        // Make hook scripts executable
+        val hookFiles = listOf("pre-commit", "pre-push", "commit-msg")
+        hookFiles.forEach { hookName ->
+            val hookFile = file("scripts/$hookName")
+            if (hookFile.exists()) {
+                hookFile.setExecutable(true)
+                println("📋 Made $hookName executable")
+            } else {
+                println("⚠️ Hook file not found: scripts/$hookName")
+            }
+        }
+
+        // Clean up old copied hooks if they exist
```

Comment:
Just to confirm: I assume this won’t cause any side effects in the future.

Reply:
Update: We've decided to remove this step. We've implemented the changes in an incremental PR.